src/router/index.js

```diff
@@ -60,4 +70,25 @@ const router = new VueRouter({
   routes
 })
 
+router.beforeEach((to, from, next) => {
+  if (!to.meta.noAuth) {
+    const token = localStorage.getItem('auth:token')
+    if (token) {
+      var data = JSON.parse(atob(token.split('.')[1]))
+      // If the token is expired, remove the token and navigate back to login
+      if (new Date(data.validUntil) > new Date()){
+        axios.defaults.headers['Authorization'] = `Bearer ${token}`
+      } else {
+        localStorage.removeItem('auth:token')
+        next('/login')
```

Comment:
Missing a return here. It will call `next('/login')` then `next()` and bug out. It would be nice if we could merge the two identical `next('/login')` paths with some trickery.